v4/install/manual.md

```diff
@@ -0,0 +1,129 @@
+---
+title: Manual Installation
+nav_order: 1
+layout: default
+parent: Installation
+grand_parent: v4
+---
+
+# Manual Installation
+In this section we will discuss how to perform a manual installation. Some requirements may link to other installation guides since we do not want to re-invent the wheel.
+
+## Prerequisites
+PoracleJS requires several prerequisites to operate. The following prerequisites must be installed for PoracleJS to function properly.
+ * [NodeJS](https://nodejs.org/en/)
+ * [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+ * A databases is required for PoracleJS to operate properly. Only one needs to be configured for PoracleJS to operate correctly.
+    * [MariaDB](https://www.digitalocean.com/community/tutorials/how-to-install-mariadb-on-ubuntu-20-04)
+    * [MySQL](https://www.digitalocean.com/community/tutorials/how-to-install-mysql-on-ubuntu-20-04) [MariaDB > MySQL :)]
+    * [PostgreSQL](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-postgresql-on-ubuntu-20-04) @TODO - Is this supported in v4?
+    * [SQLite](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-sqlite-on-ubuntu-20-04)
+   
+### Notification Service
+We will need to configure a Discord or Telegram bot to allow PoracleJS to send messages to the required service. Only one bot is required for the configuration. PoracleJS will require the token for the user regardless of the service. This will be highlighted in the individual pages on how to obtain that information.
+
+ * [Discord Bot](../../discordbot.html)
+ * [Telegram Bot](../../telegrambot.html)
+
+### Database
+A database and user need to be configured for PoracleJS to access the database. While you can reuse existing users it is recommended to create a new user for each application.
+
+#### MariaDB / MySQL
+Below are examples of how to create the database and user. Please make sure to use a [strong password](https://passwordsgenerator.net/) for the user. The example password is `poraclePassword`.
+If Poracle connects to the database locally, you can give it only local access rights:
+   ```sql
+   CREATE DATABASE poracle;
+   CREATE USER 'poracleuser'@'localhost' IDENTIFIED BY 'poraclePassword';
+   GRANT ALL PRIVILEGES ON poracle . * TO 'poracle'@'localhost';
+   exit
+   ```
+   
+Alternatively, you can grant your user access from anywhere:
+   ```sql
+   CREATE DATABASE poracle;
+   CREATE USER 'poracleuser'@'%' IDENTIFIED BY 'poraclePassword';
+   GRANT ALL PRIVILEGES ON poracle . * TO 'poracle'@'%';
+   exit
+   ```
+
```

Comment:
You may get this error with the default settings for older versions of MariaDB/MySql, 

**767 Error: alter table humans add primary key humans_pkey(id) - Specified key was too long; max key length is 767 bytes**

https://dba.stackexchange.com/questions/231219/mariadb-10-1-38-specified-key-was-too-long-max-key-length-is-767-bytes

Reply:
I ran into this issue on my system. Let me document the steps I took to solve it (pretty sure I ended up on that page anyways) 😂

Reply:
Yes, I had it when I moved to mysql under ubuntu 18 - I think the defaults in 20 are with a newer mariadb which has the new settings.  With 4-byte unicode strings it doesn't take a very long key to blast through that old index limit!